src/main/java/com/novi/backend_spring_boot_tech_it_easy_controller/services/RemoteControllerService.java

```diff
@@ -0,0 +1,93 @@
+    private final TelevisionRepository televisionRepository;
+
+    @Autowired
+    public RemoteControllerService(RemoteControllerRepository remoteControllerRepository, TelevisionRepository televisionRepository) {
+        this.remoteControllerRepository = remoteControllerRepository;
+        this.televisionRepository = televisionRepository;
+    }
+
+    public void linkRemoteControllerToTelevision(Long televisionId, Long remoteControllerId) {
+        Television television = televisionRepository.findById(televisionId)
+                .orElseThrow(() -> new RuntimeException("Television not found"));
+        RemoteController remoteController = remoteControllerRepository.findById(remoteControllerId)
+                .orElseThrow(() -> new RuntimeException("RemoteController not found"));
+
+        television.setRemoteController(remoteController);
+
+        televisionRepository.save(television);
+    }
+
+    public List<RemoteControllerDto> getAllRemoteControllers() {
+        List<RemoteController> remoteControllersList = remoteControllerRepository.findAll();
+        List<RemoteControllerDto> remoteControllerDtoList = new ArrayList<>();
+
+        for (RemoteController remoteController : remoteControllersList) {
+            RemoteControllerDto remoteControllerDto = convertToDto(remoteController);
+            remoteControllerDtoList.add(remoteControllerDto);
+        }
+
+        return remoteControllerDtoList;
+    }
+
+    public RemoteControllerDto getRemoteControllerById(Long id) {
+        RemoteController remoteController = remoteControllerRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("RemoteController not found"));
+        return convertToDto(remoteController);
+    }
+
+    public RemoteControllerDto createRemoteController(RemoteControllerInputDto remoteControllerInputDto) {
+        RemoteController remoteController = new RemoteController();
+        remoteController.setName(remoteControllerInputDto.getName());
+        remoteController.setBrand(remoteControllerInputDto.getBrand());
+        remoteController.setPrice(remoteControllerInputDto.getPrice());
+
+        remoteController = remoteControllerRepository.save(remoteController);
+        return convertToDto(remoteController);
+    }
+
+    public RemoteControllerDto updateRemoteController(Long id, RemoteControllerDto remoteControllerInputDto) {
+        RemoteController remoteController = remoteControllerRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("RemoteController not found"));
+
+        remoteController.setName(remoteControllerInputDto.getName());
+        remoteController.setBrand(remoteControllerInputDto.getBrand());
+        remoteController.setPrice(remoteControllerInputDto.getPrice());
+
+        RemoteController updatedRemoteController = remoteControllerRepository.save(remoteController);
+        return convertToDto(updatedRemoteController);
```

Comment:
Ik zie dat je mapping in je service laag begeeft. Mocht je mouwen aparte mapping component hebben zou je kunnen overwegen de mapping in de controller kunnen plaats vinden. Waardoor je service laag alleen business logica afhandeld.